src/hostapi/jack/pa_jack.c

```diff
@@ -451,6 +451,49 @@ BlockingWaitEmpty( PaStream *s )
 
 /* ---- jack driver ---- */
 
+void replace_string(char *string, char *to_find, char *replacement)
+{
+    char *found_position = strstr(string, to_find);
+    while( found_position ) {
+        size_t sizeof_original = strlen(string);
+        size_t sizeof_found = strlen(to_find);
+        size_t sizeof_replacement = strlen(replacement);
+
+        size_t sizeof_destination = sizeof_original - sizeof_found + sizeof_replacement;
+        char *buffer = (char *)malloc((sizeof_destination + 1) * sizeof(char));
+
+        size_t length_before_find = found_position - string;
+        memcpy(buffer, string, length_before_find);
+        memcpy(buffer + length_before_find, replacement, sizeof_replacement);
+        memcpy(buffer + length_before_find + sizeof_replacement,
+              string + length_before_find + sizeof_found,
+              sizeof_original - length_before_find + sizeof_found);
+        memcpy(string, buffer, sizeof_destination);
```

Comment:
This is unsave, because you potentially write behind the original string without permission.
You should take a buffer size parameter as well to be save.

Reply:
It now checks the buffer size before memmove. Is this an acceptable alternative?